### ISSUE TITLE
Concerning the TLD definition

### ISSUE THREAD

**Issue by Yomguithereal** [barseghyanartur/tld#32]
Hello @barseghyanartur. While implementing the Trie I stumbled upon something weird: in your example:

```python
get_tld("http://www.google.co.uk")
# 'google.co.uk'
```

Shouldn't the returned TLD be `co.uk` instead? Am I misunderstanding something about TLDs here? Seems you return domain origin + TLD, no?

**Comment by barseghyanartur** [comment]
@Yomguithereal:

Yep, that's True. The mistake is rather historical. In fact, ``get_tld`` returns second level domain as a result (if used without ``as_object`` flag). That change shall be fixed some day. My preference, with some transition in between). 

Perhaps, current functionality get_tld remain as get_fld (first-level-domain). And the object shall provide details.

```python
res = get_tld("http://www.google.co.uk", as_object=True)
```

**Now**

```bash
In: res.domain
Out: 'google'

In: res.extension
Out: 'co.uk'

In: res.subdomain
Out: 'www'

In: res.suffix
Out: 'co.uk'

In: res.tld
Out: 'google.co.uk'
```

**Future**

```bash
In: res.fld
Out: 'google.co.uk'

In: res.tld
Out: 'co.uk'

In: res.domain
Out: 'google'

In: res.subdomain
Out: 'www'
```

When used without ``as_object`` it shall return ``google.co.uk``. For easier migration, a new helper function ``get_fld`` shall be introduced, which will return the first level domain. If used with ``as_object`` set to True it shall return the new object (as mentioned in ``Future``).

**Comment by Yomguithereal** [comment]
I am not very accustomed to python lib packaging usages but can't we roll a breaking version through semantic versioning? Is that because most of python library consumers don't fix their deps' version and because python is unable, like npm would, to rely on more than one specific version in a dep tree?

**Comment by barseghyanartur** [comment]
@Yomguithereal:

The way I said would work excellent for everyone. It would be really easy to do something like:

```python
from tld import get_fld as get_tld
```

- ``get_tld`` shall work 100% correct.
- ``get_fld`` preserves old functionality (if not used with ``as_object`` argument set to True).

**Comment by barseghyanartur** [comment]
@Yomguithereal:

Any progress on this?

**Comment by barseghyanartur** [comment]
@Yomguithereal:

I've implemented the changes. For now, `get_fld` is the old `get_tld` (without the `as_object` argument) and `get_tld` has become a new function which acts as it should be (according to the specs specified above).

Could you check it?

**Comment by Yomguithereal** [comment]
Sure @barseghyanartur. I'll check tomorrow. We also need to speak about the `!` rule because it seems there is something wrong with it.

**Comment by Yomguithereal** [comment]
Everything looks good to me except for #39.

**Comment by barseghyanartur** [comment]
@Yomguithereal:

I'll make a release today. Issue #39 to be solved either in 0.8.x version.

**Comment by barseghyanartur** [comment]
Released in 0.8.
